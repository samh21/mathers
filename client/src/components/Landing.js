import React from 'react';
import { Link } from 'react-router-dom';
import {
  Card,
  CardMedia,
  Container,
  Grid,
  createMuiTheme,
  ThemeProvider,
  AppBar,
  Toolbar,
  Typography,
} from '@material-ui/core';
import icecream from '../img/unsplash/icecream.jpg';
import burger from '../img/unsplash/burger.jpg';
import coffee from '../img/unsplash/coffee.jpg';
import '../components/landing.css';
import GridItem from './GridItem';

export default function Landing() {
  const theme = createMuiTheme({
    spacing: 2,
  });
  return (
    // <header className="landing">
    //   <div className="dark">
    //     <h1>Mather's Ice Cream</h1>
    //     <h2>
    //       Homemade ice cream.... currently home delivery only. Free home
    //       delivery in the Forres area every Friday.
    //     </h2>
    //     <Link to="/shop" style={{ textDecoration: 'none' }}>
    //       <p className="button" href="">
    //         Enter Shop
    //       </p>
    //     </Link>
    //   </div>
    // </header>
    <>
      <ThemeProvider theme={theme}>
        <Grid container spacing={1}>
          <GridItem
            xs={12}
            image={icecream}
            title="Ice Cream"
            description="Home made with locally sourced ingredients. Dairy free and Vegan friendly options avaialble"
            action={true}
          />
          <GridItem
            xs={12}
            md={6}
            image={coffee}
            title="Coffee"
            description="Artisan coffees. Responsibly sourced beans. Roasted locally"
          />
          <GridItem
            xs={12}
            md={6}
            image={burger}
            title="Burgers"
            description="Made with 21 day aged local Beef. Vegetarian options available"
          />
        </Grid>
      </ThemeProvider>
    </>
  );
}
