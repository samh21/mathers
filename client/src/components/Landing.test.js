import React from 'react';
import Landing from './Landing';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

it('Matches snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly without crashing', () => {
  render(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>
  );
});

it('displays correctly', () => {
  const { getByText } = render(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>
  );
  getByText('Ice Cream');
  getByText('Coffee');
  getByText('Burgers');
});
