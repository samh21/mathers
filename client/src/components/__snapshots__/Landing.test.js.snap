// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<header
  className="landing"
>
  <div
    className="dark"
  >
    <h1>
      Mather's Ice Cream
    </h1>
    <h2>
      Homemade ice cream.... currently home delivery only. Free home delivery in the Forres area every Friday.
    </h2>
    <a
      href="/shop"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <p
        className="button"
        href=""
      >
        Enter Shop
      </p>
    </a>
  </div>
</header>
`;
